static/src/js/shop_list_editor.js

```diff
@@ -1,349 +1,476 @@
-	this.data.tag = this.$("#opttag").val();
-	if (this.$("input[value=num]").prop('checked')){ this.data.info = 'num' } else { this.data.info = 'km' }
-	var tagInputs = this.$("input[id*=shoptag]");
-	this.data.tags = []
-	for (var i=0; i < tagInputs.length; i++) {
-	    if (this.$(tagInputs[i]).prop('checked')) {
-		this.data.tags.push(this.$(tagInputs[i]).val())
-					      }
-						 }
-        return this._super.apply(this, arguments);
-    },
-    destroy: function () {
-        this._super.apply(this, arguments);
-    },
+        },
+    });
 
-});
+    var SourceEntryDialog = LinkDialog.extend({
+        /**
+         * Init function.
+         *
+         * @constructor
+         *
+         * @param {*} parent
+         * @param {*} options
+         * @param {*} editor
+         * @param {*} data
+         */
+        init: function (parent, options, editor, data) {
+            data.text = data.shop_list_params || '';
+            data.url = data.shop_list_url || '';
+            data.color = data.shop_list_color || '';
+            data.tag = data.shop_list_tag || '';
+            data.info = data.shop_list_info || '';
+            data.tags_avail = data.tags_avail || [];
+            data.tags = data.tags || [];
+            data.isNewWindow = data.new_window;
+            this.data = data;
+            return this._super.apply(this, arguments);
+        },
 
+        /**
+         * Start function
+         *
+         * @override
+         */
+        start: function () {
+            var self = this;
 
-options.registry.eyekraft_shop_list = options.Class.extend({
+            // remove style selector
+            this.$(".link-style").remove();
 
-    set_visibility: function (elem, visible) {
-        // save changes in edit mode
-        var $visible = $(elem).data();
-        $visible.visible = visible;
+            // remove "new window" checkbox
+            this.$(".pull-right").remove();
 
-        // save changes to object
-        elem.attr("data-visible", visible);
-    },
+            // remove selector of pages
+            this.$("ul[class=list-group]")[0].children[0].remove();
 
-    list_visibility : function (type, data) {
-        if(type !== "click") return;
-        var self = this;
-        var section = self.$target
+            this.$("h4[class=modal-title]").text(_t("Configuring the data source"));
+            this.$("label[for=link-new]").parent().replaceWith(function(index,oldHTML) {
+                return $('<h5 class="list-group-item-heading">').html(oldHTML)
+            })
+            this.$("label[for=link-new]").text(_t("Source API-key"));
+            this.$("label[for=link-external]").parent().replaceWith(function(index,oldHTML) {
+                return $('<h5 class="list-group-item-heading">').html(oldHTML)
+            })
+            this.$("label[for=link-external]").text(_t("Source URL"));
+            var mainDiv = this.$("div[class=list-group]").parent();
+            var divListGroup = $(
```

Comment:
Can be defined as a template in xml file and parsed here as via `qweb` from Odoo JS Core